src/v0/sources/shopify/util.js

```diff
@@ -190,14 +183,37 @@ const setAnonymousIdorUserIdFromDb = async (message, metricMetadata) => {
     })
     anonymousIDfromDB = sha256(cartToken).toString().substring(0, 36);
   }
-  message.setProperty('anonymousId', anonymousIDfromDB);
+  return anonymousIDfromDB;
 };
 
+/**
+ * It checks if the event is valid or not based on previous cartItems
+ * @param {*} inputEvent 
+ * @returns true if event is valid else false
+ */
+const isValidCartEvent = async (inputEvent) => {
+  const cartToken = inputEvent.token || inputEvent.id;
+  const redisVal = await RedisDB.getVal(cartToken);
+  const newCartItems = inputEvent?.line_items.length !== 0 ? sha256(inputEvent.line_items) : 0;
+  if (redisVal) {
+    const prevCartItems = redisVal.itemsHash;
+    if (prevCartItems === newCartItems) {
+      return false;
+    }
+    await anonymousIdCache.get(cartToken, () => redisVal.anonymousId);
```

Comment:
Shouldn't it be set?

Reply:
its from cache class. We don't have set function but instead we pass a fallback function through which we can set the val. Ref:
https://github.com/rudderlabs/rudder-transformer/blob/383a42a26873e03320d7abbeabaa0dea5856873b/src/v0/util/cache.js#L12